rust/headless-client/src/heartbeat.rs

```diff
@@ -0,0 +1,56 @@
+//! A heartbeat that logs to `INFO` at exponentially increasing intervals
```

Comment:
Do you mind adding more details explaining what this is for?

Reply:
If what you want is to have some notion of the time for the logs you have [this option](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/struct.SubscriberBuilder.html#method.with_timer) for the default subscriber

Reply:
After adding the IPC service, the GUI logs are very quiet when the Client is in a steady state, (and the IPC service is very quiet when the GUI is closed / signed out) so I wanted something that gives an approximate idea of how long the processes stayed up without completely hanging or crashing.

Reply:
For crashes we can add something in the panic hook.

For hanging, I'm not sure an exponential heartbeat is ideal, it only give us a lower bound. What can we do with this information?

Reply:
I was thinking if it's a laptop and the battery dies / user holds the power button then we won't get anything logged.
For the same reason as checking the system uptime I'm just curious if the user restarts or something and that fixes an issue, or if they let it run for a while.

I think it gives us an upper bound too, though after a day or two of uptime it'll be measured in days.

Reply:
I don't think we have an upper bound, without counting on connlib-side logging something. If we see "has been up for 2 days" it means it has been up for somewhere between 2 to 4 days or have hanged before 4 days.

Reply:
Hm isn't 4 days the upper bound then? If we maxed out at 1 heartbeat per day we still wouldn't be sure if it had hung until the next heartbeat

Reply:
I mean, yes in general, I'm not sure how you would distinguish between the process being hung or the time haven't elapsed

Reply:
It's sort of a variation of the "Detect a network loss without sending any packets" problem, I think you just can't

Reply:
yeah, that's what I mean, I'm just not sure these logs are useful

Reply:
anyways, if we decide to keep it I think we can simplify the `heartbeat` function to 
```rs
pub async fn heartbeat() {
    let mut next_heartbeat = Duration::from_secs(1);
    loop {
        sleep(next_heartbeat).await;
        tracing::info!(?system_uptime, "Heartbeat");
        next_heartbeat *= 2;
    }
}
```

we don't use the configuration

Reply:
We could, since it's tested, I had that struct broken out so I could prove it wouldn't overflow, and used `sleep_until` so it hopefully won't drift